### ISSUE TITLE
Add option for block indentation of array literals

### ISSUE THREAD

**Issue by Ms2ger** [rust-lang-nursery/rustfmt#574]
```diff
-static ERROR_FORMAT_STRING_STRING: [libc::c_char; 4] = [
-    '{' as libc::c_char,
-    '0' as libc::c_char,
-    '}' as libc::c_char,
-    0 as libc::c_char,
-];
+static ERROR_FORMAT_STRING_STRING: [libc::c_char; 4] = ['{' as libc::c_char,
+                                                        '0' as libc::c_char,
+                                                        '}' as libc::c_char,
+                                                        0 as libc::c_char];
```

is not an improvement at all.

**Comment by nrc** [comment]
Looks OK to me - it's consistent with how we would format shorter arrays or functions, etc. Even if we did prefer this style here (I can sympathise that the large block of whitespace is sub-optimal) it's hard to imagine how you would choose between visual and block indent.

We should have an option for this I guess (since we do for most other places where you could choose either way). Changing the title accordingly.

**Comment by slyrz** [comment]
> consistent with how we would format shorter arrays or functions, etc

But not consistent with the initialization of structs. 

The current output might work with short variable names, short type signatures and few array items. Once any of these get larger, the formatting becomes problematic. Think of cryptographic algorithms where you have to declare large blocks of 64 bit constants. With one item per line almost flushed to the right, the source code will look very strange.

**Comment by hsivonen** [comment]
I suggest adding a line break after `[` in large array literals. [Example of non-ideal formatting](https://github.com/hsivonen/encoding-rs/blob/9ae908b0f0132f562c506c8bd7e2f9772c6520f0/src/data.rs#L13).

**Comment by marcusklaas** [comment]
This was addressed in #893.
